### PR TITLE
clean up /components

### PR DESCRIPTION
**Pull request by alexisintech**
- removes unnecessary imports
- migrates `<Table />` components to markdown tables
- adds links to references where necessary

### CODE REVIEW

File: docs/components/authentication/sign-in.mdx [hunk 1]
```diff
@@ -99,12 +99,15 @@ All props below are optional.
 
 | Name | Type | Description |
 | --- | --- | --- |
-| `appearance` | `object` | Controls the overall look and feel of the component. See [Appearance](/docs/components/customization/overview) for more information. |
-| `routing` | `string` | The routing strategy for your pages.<br />Supported values are: <ul><li>hash (default): Hash-based routing.</li><li>path: Path-based routing.</li><li>virtual: Virtual-based routing.</li></ul> |
-| `path` | `string` | The path where the component is mounted on when path-based routing is used e.g. /sign-up. Note: If you are using Environment Variables for Next.js or Remix to specify your routes, this will be set to `path`.|
```
Comment (BRKalow) [inline docs/components/authentication/sign-in.mdx:104]: Oops it looks like this note was on the wrong prop. We should keep the note but move it to the `routing` row.

File: docs/components/authentication/sign-up.mdx [hunk 2]
```diff
@@ -100,8 +100,8 @@ All props below are optional.
 
 | Name | Type | Description |
 | --- | --- | --- |
-| `appearance` | `object` | Controls the overall look and feel of the component. See [Appearance](/docs/components/customization/overview) for more information. |
-| `routing` | `string` | The routing strategy for your pages.<br />Supported values are: <ul><li>hash (default): Hash-based routing.</li> <li>path: Path-based routing.</li> <li>virtual: Virtual-based routing</li></ul>Note: If you are using Environment Variables for Next.js or Remix to specify your routes, this will be set to `path`.|
```
Comment (BRKalow) [inline docs/components/authentication/sign-up.mdx:104]: This note should stay 👍

File: docs/components/clerk-provider.mdx [hunk 3]
```diff
@@ -81,80 +77,18 @@ ReactDOM.render(
...
-        <code>afterSignUpUrl</code>,
-        <code>string</code>,
-        <>Full URL or path to navigate after successful sign up.</>,
-      ],
-    }
-  ]}
-/>
+| Name | Type | Description |
+| --- | --- | --- |
+| `publishableKey` | `string \| undefined` | The publishable key from your Clerk Dashboard, used to connect to Clerk |
```
Comment (BRKalow) [inline docs/components/clerk-provider.mdx:82]:
```suggestion
| `publishableKey` | `string \| undefined` | The publishable key from your Clerk Dashboard, used to connect to Clerk. |

```

File: docs/components/customization/layout.mdx [hunk 4]
```diff
@@ -31,98 +27,13 @@ import { ClerkProvider } from '@clerk/nextjs';
...
+| Name | Type | Description |
+| --- | --- | --- |
+| `helpPageUrl` | `string` | The URL to your help page |
+| `logoImageUrl` | `string` | The URL to your logo image |
+| `logoPlacement` | `'inside' \| 'outside'` | The placement of your logo.  |
+| `privacyPageUrl` | `string` | The URL to your privacy page |
+| `showOptionalFields` | `boolean` | Whether to show optional fields on sign up. |
+| `socialButtonsPlacement` | `'bottom' \| 'top'` | The placement of your social buttons. |
+| `socialButtonsVariant` | `'blockButton' \| 'textButton' \| 'auto'` | The variant of your social buttons. |
+| `termsPageUrl` | `string` | The URL to your terms page |
```
Comment (BRKalow) [inline docs/components/customization/layout.mdx:39]:
```suggestion
| `helpPageUrl` | `string` | The URL to your help page. |
| `logoImageUrl` | `string` | The URL to your logo image. |
| `logoPlacement` | `'inside' \| 'outside'` | The placement of your logo.  |
| `privacyPageUrl` | `string` | The URL to your privacy page. |
| `showOptionalFields` | `boolean` | Whether to show optional fields on sign up. |
| `socialButtonsPlacement` | `'bottom' \| 'top'` | The placement of your social buttons. |
| `socialButtonsVariant` | `'blockButton' \| 'textButton' \| 'auto'` | The variant of your social buttons. |
| `termsPageUrl` | `string` | The URL to your terms page. |
```

File: docs/components/customization/variables.mdx [hunk 5]
```diff
@@ -28,100 +24,22 @@ import { ClerkProvider } from '@clerk/nextjs';
...
+| `colorBackground` | `string` | The background color for the card container |
+| `colorInputText` | `string` | The color used for text in input fields |
+| `colorInputBackground` | `string` | The background color used for input fields |
+| `fontFamily` | `string` | The font family used throughout the components. By default, it is inherited. |
+| `fontFamilyButtons` | `string` | The font family used for buttons. By default, it is inherited. |
+| `fontSize` | `string` | The font size used throughout the components. By default, this is set to 1rem. |
+| `fontSmoothing` | `'auto' \| 'antialiased' \| 'never'` | The font smoothing used throughout the components. By default, this is set to auto. |
+| `fontWeight` | `{normal: number, medium: number, bold: number}` | The font weight used throughout the components. By default, this is set to `{normal: 400, medium: 500, bold: 600}` |
+| `borderRadius` | `string` | The border radius used throughout the components. By default, this is set to 0.375rem |
+| `spacingUnit` | `string` | The spacing unit used throughout the components. By default, this is set to 1rem. |
```
Comment (BRKalow) [inline docs/components/customization/variables.mdx:45]:
```suggestion
| `colorPrimary` | `string` | The primary color used throughout the components. |
| `colorDanger` | `string` | The color used for error states. |
| `colorSuccess` | `string` | The color used for success states. |
| `colorWarning` | `string` | The color used for warning states. |
| `colorAlphaShade` | `string` | The color that will be used for all to generate the alpha shades the components use. This option applies to borders, backgrounds for hovered elements, hovered dropdown options. |
| `colorTextOnPrimaryBackground` | `string` | The color used for text on the primary background. |
| `colorTextSecondary` | `string` | The color used for secondary text. |
| `colorBackground` | `string` | The background color for the card container. |
| `colorInputText` | `string` | The color used for text in input fields. |
| `colorInputBackground` | `string` | The background color used for input fields. |
| `fontFamily` | `string` | The font family used throughout the components. By default, it is inherited. |
| `fontFamilyButtons` | `string` | The font family used for buttons. By default, it is inherited. |
| `fontSize` | `string` | The font size used throughout the components. By default, this is set to 1rem. |
| `fontSmoothing` | `'auto' \| 'antialiased' \| 'never'` | The font smoothing used throughout the components. By default, this is set to auto. |
| `fontWeight` | `{normal: number, medium: number, bold: number}` | The font weight used throughout the components. By default, this is set to `{normal: 400, medium: 500, bold: 600}`. |
| `borderRadius` | `string` | The border radius used throughout the components. By default, this is set to `0.375rem`. |
| `spacingUnit` | `string` | The spacing unit used throughout the components. By default, this is set to `1rem`. |
```

File: docs/components/organization/organization-list.mdx [hunk 6]
```diff
@@ -94,11 +94,13 @@ All props below are optional.
...
-| `apperance` | `object` | Controls the overall look and feel.
-| `afterCreateOrganizationUrl` | <code>((org: [Organization][org-ref]) => string)</code> \| `string` | Full URL or path to navigate after creating a new organization.<br />Defaults to `undefined`.
-| `afterSelectOrganizationUrl` | <code>((org: [Organization][org-ref]) => string)</code> \| `string` | Full URL or path to navigate after selecting an organization.<br />Defaults to `undefined`.
-| `afterSelectPersonalUrl` | <code>((org: [Organization][org-ref]) => string)</code> \| `string` | Full URL or path to navigate after selecting the personal account.<br />Defaults to `undefined`.
+| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether `<OrganizationList />` will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations. <br />Defaults to `false` |
```
Comment (BRKalow) [inline docs/components/organization/organization-list.mdx:97]:
```suggestion
| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether `<OrganizationList />` will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations. <br />Defaults to `false`. |
```

File: docs/components/organization/organization-switcher.mdx [hunk 7]
```diff
@@ -79,97 +79,21 @@ The `<OrganizationSwitcher />` component is used to enable the ability to switch
...
-    },
-  ]}
-/>
+| Name | Type | Description |
+| --- | --- | --- |
+| `afterCreateOrganizationUrl` | `string` | Full URL or path to navigate after creating a new organization. |
+| `appearance` | <code>[Appearance](/docs/components/customization/overview) \| undefined</code> | Optional object to style your components. Will only affect [Clerk Components][components-ref] and not [Account Portal][ap-ref] pages. |
+| `createOrganizationUrl` | `string` | Full URL or path where the [`<CreateOrganization />`][createorg-ref] component is mounted. |
+| `organizationProfileUrl` | `string` | Full URL or path where the [`<OrganizationProfile />`][orgprofile-ref] component is mounted. |
+| `createOrganizationMode` | `'modal' \| 'navigation'` | Controls whether clicking the "Create organization" button will cause the [`<CreateOrganization />`][createorg-ref] component to open as a modal, or if the browser will navigate to the `createOrganizationUrl` where [`<CreateOrganization />`][createorg-ref] is mounted as a page.<br />Defaults to: `"modal"` |
```
Comment (BRKalow) [inline docs/components/organization/organization-switcher.mdx:88]:
```suggestion
| `createOrganizationMode` | `'modal' \| 'navigation'` | Controls whether clicking the "Create organization" button will cause the [`<CreateOrganization />`][createorg-ref] component to open as a modal, or if the browser will navigate to the `createOrganizationUrl` where [`<CreateOrganization />`][createorg-ref] is mounted as a page.<br />Defaults to: `"modal"`. |
```

File: docs/components/organization/organization-switcher.mdx [hunk 8]
```diff
@@ -79,97 +79,21 @@ The `<OrganizationSwitcher />` component is used to enable the ability to switch
...
+| `organizationProfileMode` | `'modal' \| 'navigation'` | Controls whether clicking the "Manage organization" button will cause the [`<OrganizationProfile />`][orgprofile-ref] component to open as a modal, or if the browser will navigate to the `organizationProfileUrl` where [`<OrganizationProfile />`][orgprofile-ref] is mounted as a page.<br />Defaults to: `"modal"` |
```
Comment (BRKalow) [inline docs/components/organization/organization-switcher.mdx:89]:
```suggestion
| `organizationProfileMode` | `'modal' \| 'navigation'` | Controls whether clicking the "Manage organization" button will cause the [`<OrganizationProfile />`][orgprofile-ref] component to open as a modal, or if the browser will navigate to the `organizationProfileUrl` where [`<OrganizationProfile />`][orgprofile-ref] is mounted as a page.<br />Defaults to: `"modal"`. |
```

File: docs/components/organization/organization-switcher.mdx [hunk 9]
```diff
@@ -79,97 +79,21 @@ The `<OrganizationSwitcher />` component is used to enable the ability to switch
...
+| `afterLeaveOrganizationUrl` | `string` | Full URL or path to navigate to after the user leaves the currently active organization. |
+| `afterSwitchOrganizationUrl` | `string` | Full URL or path to navigate after a successful organization switch. |
+| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether [`<OrganizationSwitcher />`][orgswitcher-ref] will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations.<br />Defaults to: `true` |
```
Comment (BRKalow) [inline docs/components/organization/organization-switcher.mdx:92]:
```suggestion
| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether [`<OrganizationSwitcher />`][orgswitcher-ref] will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations.<br />Defaults to: `true`. |
```

File: docs/components/user/user-button.mdx [hunk 10]
```diff
@@ -200,83 +196,19 @@ In this example, `<UserButton />` is mounted inside a header component, which is
...
-      ],
-    },
-  ]}
-/>
+| Name | Type | Description |
+| --- | --- | --- |
+| `appearance` | <code>[Appearance](/docs/components/customization/overview) \| undefined</code> | Optional object to style your components. Will only affect [Clerk Components][components-ref] and not [Account Portal][ap-ref] pages. |
+| `showName` | `boolean` | Controls if the user name is displayed next to the user image button. |
+| `signInUrl` | `string` | The full URL or path to navigate to when the "Add another account" button is clicked. |
+| `userProfileMode` | <code>"modal" \| "navigation"</code> | Controls whether clicking the "Manage your account" button will cause the [`<UserProfile />`][userprofile-ref] component to open as a modal, or if the browser will navigate to the `userProfileUrl` where [`<UserProfile />`][userprofile-ref] is mounted as a page.<br />Defaults to: `"modal"` |
```
Comment (BRKalow) [inline docs/components/user/user-button.mdx:204]:
```suggestion
| `userProfileMode` | <code>"modal" \| "navigation"</code> | Controls whether clicking the "Manage your account" button will cause the [`<UserProfile />`][userprofile-ref] component to open as a modal, or if the browser will navigate to the `userProfileUrl` where [`<UserProfile />`][userprofile-ref] is mounted as a page.<br />Defaults to: `"modal"`. |
```

File: docs/components/user/user-button.mdx [hunk 11]
```diff
@@ -200,83 +196,19 @@ In this example, `<UserButton />` is mounted inside a header component, which is
...
+| `userProfileUrl` | `string` | The full URL or path leading to the user management interface. |
+| `afterSignOutUrl` | `string` | The full URL or path to navigate to after a signing out from all accounts (applies to both single-session and multi-session apps) |
```
Comment (BRKalow) [inline docs/components/user/user-button.mdx:206]:
```suggestion
| `afterSignOutUrl` | `string` | The full URL or path to navigate to after a signing out from all accounts (applies to both single-session and multi-session apps). |
```

File: docs/components/user/user-button.mdx [hunk 12]
```diff
@@ -200,83 +196,19 @@ In this example, `<UserButton />` is mounted inside a header component, which is
...
+| `afterMultiSessionSingleSignOutUrl` | `string` | The full URL or path to navigate to after a signing out from currently active account (multisession apps). |
+| `afterSwitchSessionUrl` | `string` | The full URL or path to navigate to after a successful account change (multi-session apps). |
+| `defaultOpen` | `boolean` | Controls whether the `<UserButton />` should open by default during the first render. |
+| `userProfileProps` | `object` | Specify options for the underlying [`<UserProfile />`][userprofile-ref] component. e.g. `{additionalOAuthScopes: {google: ['foo', 'bar'], github: ['qux']}}` |
```
Comment (BRKalow) [inline docs/components/user/user-button.mdx:210]:
```suggestion
| `userProfileProps` | `object` | Specify options for the underlying [`<UserProfile />`][userprofile-ref] component. e.g. `{additionalOAuthScopes: {google: ['foo', 'bar'], github: ['qux']}}`. |
```

File: docs/components/user/user-profile.mdx [hunk 13]
```diff
@@ -72,44 +72,12 @@ The `<UserProfile />` component is used to render a beautiful, full-featured acc
...
-      ],
-    },
-  ]}
-/>
+| Name | Type | Description |
+| --- | --- | --- |
+| `appearance` | <code>[Appearance](/docs/components/customization/overview) \| undefined</code> | Optional object to style your components. Will only affect [Clerk Components][components-ref] and not [Account Portal][ap-ref] pages. |
+| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages. <br />Supported values are: <ul><li>`hash` (default): Hash-based routing.</li> <li>`path`: Path-based routing.</li> <li>`virtual`: Virtual-based routing.</li></ul> |
+| `path` | `string` | The path where the component is mounted on when path-based routing is used e.g. /sign-in. |
+| `additionalOAuthScopes` | `object` | Specify additional scopes per OAuth provider that your users would like to provide if not already approved. <br />e.g. `{google: ['foo', 'bar'], github: ['qux']}` |
```
Comment (BRKalow) [inline docs/components/user/user-profile.mdx:80]:
```suggestion
| `additionalOAuthScopes` | `object` | Specify additional scopes per OAuth provider that your users would like to provide if not already approved. <br />e.g. `{google: ['foo', 'bar'], github: ['qux']}`. |
```

File: docs/references/javascript/clerk/sign-up.mdx [hunk 14]
```diff
@@ -250,7 +250,6 @@ All props below are optional.
 | `afterSignInUrl` | `string` | The full URL or path to navigate after a successful sign in. |
 | `signInUrl` | `string` | Full URL or path to the sign in page. Use this property to provide the target of the 'Sign In' link that's rendered. |
 | `afterSignUpUrl` | `string` | The full URL or path to navigate after a successful sign up. |
-| `afterSignUpUrl?` | `string` | The full URL or path to navigate after a successful sign up. |
```
Comment (BRKalow) [inline docs/references/javascript/clerk/sign-up.mdx:253]: Good catch!

**Review by BRKalow — CHANGES_REQUESTED** [review]
Some very minor formatting things, but overall looks great!

**Review by jescalan — APPROVED** [review]
LGTM, but note that i didnt go through every page of the preview and check to make sure it looks right, and want to be sure that someone has done so before merge

File: docs/references/javascript/clerk/create-organization.mdx [hunk 15]
```diff
@@ -243,7 +243,7 @@ All props below are optional.
 | Name | Type | Description |
 | --- | --- | --- |
 | `afterCreateOrganizationUrl` | `string` | Full URL or path to navigate after creating a new organization. |
-| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages. <br />Supported values are: <ul><li>`hash` (default): Hash-based routing.</li> <li>`path`: Path-based routing.</li> <li>`virtual`: Virtual-based routing.</li></ul> |
+| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages.<br />Note: If you are using environment variables for Next.js or Remix to specify your routes, this will be set to `path`. |
```
Comment (BRKalow) [inline docs/references/javascript/clerk/create-organization.mdx:246]: I don't think this note is applicable to to this component. AFAIK only sign in and sign up are impacted.

File: docs/references/javascript/clerk/organization-profile.mdx [hunk 16]
```diff
@@ -243,7 +243,7 @@ All props below are optional.
 | Name | Type | Description |
 | --- | --- | --- |
 | `afterLeaveOrganizationUrl` | `string` | Full URL or path to navigate after leaving an organization. |
-| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages. <br />Supported values are: <ul><li>`hash` (default): Hash-based routing.</li> <li>`path`: Path-based routing.</li> <li>`virtual`: Virtual-based routing.</li></ul> |
+| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages.<br />Note: If you are using environment variables for Next.js or Remix to specify your routes, this will be set to `path`. |
```
Comment (BRKalow) [inline docs/references/javascript/clerk/organization-profile.mdx:246]: I don't think this note is applicable to to this component. AFAIK only sign in and sign up are impacted.

File: docs/references/javascript/clerk/user-profile.mdx [hunk 17]
```diff
@@ -243,10 +243,10 @@ All props below are optional.
 | Name | Type | Description |
 | --- | --- | --- |
 | `appearance` | <code>[Appearance](/docs/components/customization/overview) \| undefined</code> | Optional object to style your components. Will only affect [Clerk Components][components-ref] and not [Account Portal][ap-ref] pages. |
-| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages. <br />Supported values are: <ul><li>`hash` (default): Hash-based routing.</li> <li>`path`: Path-based routing.</li> <li>`virtual`: Virtual-based routing.</li></ul> |
+| `routing` | `'hash' \| 'path' \| 'virtual'` | The routing strategy for your pages.<br />Note: If you are using environment variables for Next.js or Remix to specify your routes, this will be set to `path`. |
```
Comment (BRKalow) [inline docs/references/javascript/clerk/user-profile.mdx:246]: I don't think this note is applicable to to this component. AFAIK only sign in and sign up are impacted.

**Review by BRKalow — APPROVED** [review]

**Pull request merged**